spec/rspec/matchers/built_in/yield_spec.rb

```diff
@@ -628,6 +695,15 @@ def each_arg(*args, &block)
         |         got: ["food", "barn"]
       EOS
     end
+
+    it 'fails when the successively yielded args match the matchers at return time only' do
+      expect { |b|
+        %w[ barn food ].each do |val|
+          _yield_with_args(val, &b)
+          val.delete!(val)
+        end
+      }.not_to yield_successive_args(a_string_matching(/foo/), a_string_matching(/bar/))
+    end
```

Comment:
This test also passes both on master and this branch, so it's not actually testing it properly.  Also, as your doc string says, you're expecting failure in this case, but you haven't wrapped it in `expect { ... }.to fail_with(...)` to express that it should fail with an expectation.

As above, this also needs to swap the order of either expected or actual to force return vs yield time to come into play.  Here's what I came up with:

``` ruby
it 'fails when the successively yielded args match the matchers at return time only' do
  expect {
    expect { |b|
      %w[ food barn ].each do |val|
        _yield_with_args(val, &b)
        val.delete!(val)
      end
    }.not_to yield_successive_args(a_string_matching(/foo/), a_string_matching(/bar/))
  }.to fail_with(dedent <<-EOS)
    |expected given block not to yield successively with arguments, but yielded with expected arguments
    |expected not: [(a string matching /foo/), (a string matching /bar/)]
    |         got: ["", ""]
  EOS
end
```

I *think* that's right, and it fails on master but passes on your branch.

(And yes, this is quite confusing!).